scipy/stats/_continuous_distns.py

```diff
@@ -5959,6 +5959,48 @@ def _entropy(self):
 semicircular = semicircular_gen(a=-1.0, b=1.0, name="semicircular")
 
 
+class sine_gen(rv_continuous):
+    r"""A sine continuous random variable.
+
+    %(before_notes)s
+
+    Notes
+    -----
+    The sine distribution is useful for polar angles.
+    The probability density function for `sine` is:
+
+    .. math::
+
+        f(x) = \frac{1}{2} \sin(x)
+
+    for :math:`0 \le x \le \pi`.
+
+    %(after_notes)s
+
+    %(example)s
+
+    """
+    def _pdf(self, x):
+        # sin.pdf(x) = 1/2 * sin(x)
+        return 1.0/2*(np.sin(x))
+
+    def _rvs(self):
+        z = self._random_state.uniform(size=self._size)
+        return 2*np.arcsin(np.sqrt(z))
```

Comment:
I don't understand where the sqrt term comes from. Simply inverting the cdf gives arccos(...).

Reply:
`2*arcsin(sqrt(x)) = arccos(1-2*x)`. Re-written as the latter